lib/firewall/firewall.cc

```diff
@@ -586,6 +586,55 @@ namespace
 		uint16_t destinationPort;
 	} __packed;
 
+	struct TCPHeader
+	{
+		/**
+		 * Source port.
+		 */
+		uint16_t sourcePort;
+		/**
+		 * Destination port.
+		 */
+		uint16_t destinationPort;
+		/**
+		 * Sequence number.
+		 */
+		uint32_t sequenceNumber;
+		/**
+		 * Acknowledgement number.
+		 */
+		uint32_t acknowledgementNumber;
+		/**
+		 * Reserved bits, data offset, and flags.
+		 */
+		uint16_t bitfield;
+		/**
+		 * Window size.
+		 */
+		uint16_t windowSize;
+		/**
+		 * Checksum.
+		 */
+		uint16_t checksum;
+		/**
+		 * Urgent pointer.
+		 */
+		uint16_t urgentPointer;
+	} __packed;
```

Comment:
Just for paranoia, a `_Static_assert(sizeof(TCPHeader) == 20)`, perhaps.  (And maybe move the one for the `IPv4Header` below up above `TCPHeader`?)